CRM/Member/BAO/Membership.php

```diff
@@ -374,8 +359,8 @@ public static function create(&$params, $ids = []) {
           $params['line_item'] = $params['lineItems'];
         }
         // do cleanup line items if membership edit the Membership type.
-        if (!empty($ids['membership'])) {
-          CRM_Price_BAO_LineItem::deleteLineItems($ids['membership'], 'civicrm_membership');
+        if (!empty($params['membership_id'])) {
```

Comment:
Did you dig into the impact here - this feels like code that might be removed rahter than broadened?